### PR TITLE
Add trafilatura to the readme

### PR DESCRIPTION
**Pull request by lopuhin**
Add trafilatura from #4 

@adbar please tell if you have any suggestions for how to improve description or formatting. Thanks again for an impressive contribution.

### CODE REVIEW

**Comment by adbar** [comment]
Nice, thank you!

Feel free to run trafilatura locally on your benchmark if you haven't already done it.

A question on your _accuracy_ measure: how come it's so low for all systems? It should be closer to the F1 than that, or you're actually performing another type of calculation that isn't described explicitly.

File: README.rst [hunk 1]
```diff
@@ -28,6 +28,12 @@ Results of the evaluation (see below for reproduction and details)::
     readability          precision=0.913 ± 0.014  recall=0.931 ± 0.016  F1=0.922 ± 0.014 accuracy=0.315 ± 0.035
     xpath-text           precision=0.246 ± 0.016  recall=0.992 ± 0.001  F1=0.394 ± 0.020 accuracy=0.000 ± 0.000
 
+Result of packages added after original evaluation::
+
+    trafilatura          precision=0.925 ± 0.011  recall=0.966 ± 0.009  F1=0.945 ± 0.009 accuracy=0.221 ± 0.031
+
+Below you can find more deatails about the packages and result reproduction.
```
Comment (adbar) [inline README.rst]: should be: _details_

**Comment by lopuhin** [comment]
> Feel free to run trafilatura locally on your benchmark if you haven't already done it.

Yep, already did this.

> A question on your accuracy measure: how come it's so low for all systems? It should be closer to the F1 than that, or you're actually performing another type of calculation that isn't described explicitly.

Right it's much lower - there are more details in the PDF https://github.com/scrapinghub/article-extraction-benchmark/releases/download/v1.0.0/paper-v1.0.0.pdf but the main reason is that to compute accuracy, we take the full article text (after normalization) and check if it's exactly equal to the reference. So even minor difference will set it to 0 for a particular article, e.g. some small bit of unrelated text or a small bit of missed text. While to compute F1/precision/recall we use 4-grams which are much more forgiving to small mistakes. So it's quite noisy and also low.

**Comment by adbar** [comment]
I see!
Why 4-grams (shingles right?) and not 3- or 5-grams? Did it seem to give a better impression overall?

**Comment by adbar** [comment]
Autoextract is your system, the one you're selling API access for right? Maybe you could also explain in the readme what Diffbot is, it isn't clear.

**Comment by adbar** [comment]
2 systems (Autoextract and Diffbot) are commercial services while the rest is open-source? You could maybe add a sentence to explain it.

**Comment by adbar** [comment]
Another idea: did you install `cchardet`? (or `trafilatura[all]`) This could improve results slightly for pages with non-latin encodings, at least it does on my data. Anyway, that's all for now for my feedback, you can merge the PR if you wish ;)

File: README.rst [hunk 2]
```diff
@@ -28,6 +28,12 @@ Results of the evaluation (see below for reproduction and details)::
     readability          precision=0.913 ± 0.014  recall=0.931 ± 0.016  F1=0.922 ± 0.014 accuracy=0.315 ± 0.035
     xpath-text           precision=0.246 ± 0.016  recall=0.992 ± 0.001  F1=0.394 ± 0.020 accuracy=0.000 ± 0.000
 
+Result of packages added after original evaluation::
```
Comment (kmike) [inline README.rst:40]: I wonder if trafilatura should be also added to the first paragraph of README ("Article extraction benchmark: open-source libraries and commercial services" section)

Comment (lopuhin) [reply README.rst:40]: thanks for the catch @kmike , done in f5f76dc

**Comment by lopuhin** [comment]
> Why 4-grams (shingles right?) and not 3- or 5-grams? Did it seem to give a better impression overall?

3 seemed a bit low as it could produce false matches, and there is little difference between 4 and 5 (see below), but there is no scientific basis behind this particular value:

default (4-grams):

```
AutoExtract          precision=0.984 ± 0.003  recall=0.956 ± 0.010  F1=0.970 ± 0.005 accuracy=0.470 ± 0.037 
Diffbot              precision=0.958 ± 0.009  recall=0.944 ± 0.013  F1=0.951 ± 0.010 accuracy=0.348 ± 0.036 
boilerpipe           precision=0.850 ± 0.015  recall=0.870 ± 0.020  F1=0.860 ± 0.016 accuracy=0.006 ± 0.005 
dragnet              precision=0.925 ± 0.012  recall=0.889 ± 0.018  F1=0.907 ± 0.013 accuracy=0.221 ± 0.032 
html-text            precision=0.500 ± 0.017  recall=0.994 ± 0.001  F1=0.665 ± 0.015 accuracy=0.000 ± 0.000 
newspaper            precision=0.917 ± 0.013  recall=0.906 ± 0.018  F1=0.912 ± 0.014 accuracy=0.260 ± 0.032 
readability          precision=0.913 ± 0.014  recall=0.931 ± 0.015  F1=0.922 ± 0.013 accuracy=0.315 ± 0.035 
trafilatura          precision=0.925 ± 0.011  recall=0.966 ± 0.010  F1=0.945 ± 0.009 accuracy=0.221 ± 0.031 
xpath-text           precision=0.246 ± 0.015  recall=0.992 ± 0.001  F1=0.394 ± 0.019 accuracy=0.000 ± 0.000 
```

5-grams:

```
AutoExtract          precision=0.984 ± 0.003  recall=0.955 ± 0.010  F1=0.969 ± 0.006 accuracy=0.470 ± 0.037 
Diffbot              precision=0.956 ± 0.008  recall=0.942 ± 0.013  F1=0.949 ± 0.010 accuracy=0.348 ± 0.036 
boilerpipe           precision=0.848 ± 0.016  recall=0.868 ± 0.020  F1=0.858 ± 0.016 accuracy=0.006 ± 0.006 
dragnet              precision=0.922 ± 0.012  recall=0.887 ± 0.018  F1=0.904 ± 0.013 accuracy=0.221 ± 0.029 
html-text            precision=0.498 ± 0.018  recall=0.992 ± 0.001  F1=0.663 ± 0.016 accuracy=0.000 ± 0.000 
newspaper            precision=0.914 ± 0.013  recall=0.904 ± 0.017  F1=0.909 ± 0.014 accuracy=0.260 ± 0.034 
readability          precision=0.911 ± 0.014  recall=0.930 ± 0.015  F1=0.920 ± 0.013 accuracy=0.315 ± 0.034 
trafilatura          precision=0.923 ± 0.011  recall=0.965 ± 0.009  F1=0.943 ± 0.009 accuracy=0.221 ± 0.031 
xpath-text           precision=0.245 ± 0.015  recall=0.989 ± 0.001  F1=0.393 ± 0.019 accuracy=0.000 ± 0.000
```
> Another idea: did you install cchardet? (or trafilatura[all]) This could improve results slightly for pages with non-latin encodings, at least it does on my data.

good point, thanks - chardet was already installed, I re-run with cchardet langid installed and the output was the same.

> Autoextract is your system, the one you're selling API access for right? Maybe you could also explain in the readme what Diffbot is, it isn't clear.
> 2 systems (Autoextract and Diffbot) are commercial services while the rest is open-source? You could maybe add a sentence to explain it.

Thanks for feedback, mentioned this at the top of the README in f5f76dc

File: README.rst (hunk 1 above)
Comment (lopuhin) [reply README.rst]: thanks, fixed

**Comment by lopuhin** [comment]
thanks for feedback @adbar and @kmike , I merged the PR, happy to make more adjustments if needed.

**Pull request merged**